TempMail/utils/Email.swift

```diff
@@ -35,10 +41,31 @@ class Email : ObservableObject{
         return String((0..<8).map{ _ in letters.randomElement()!})
     }
     
+    //parses JSON data
+    private func parseData(data: Data, type: DataType) -> (inbox: [InboxModel]?, msg: MessageModel?) {
+        do {
+            //will execute the code depending on what type of data we are parsing
+            switch type {
+            case DataType.inbox:
+                var response: [InboxModel]?
+                response = try JSONDecoder().decode([InboxModel].self, from: data)
+                return(response,nil)
+            case DataType.msg:
+                var response: MessageModel?
+                response = try JSONDecoder().decode(MessageModel.self, from: data)
+                return(nil,response)
+            }
+        } catch {
+            print("Fetch failed: \(error.localizedDescription)")
+        }
+        //will never reach here
+        return (nil,nil)
```

Comment:
It gets all nasty when I try to throw an error. I think I have to make another event handler, but since I call this method alot I have to do that in each func I call it in. I think it's more trouble than its worth.